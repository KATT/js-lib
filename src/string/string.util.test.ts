import { HttpError } from '..'
import {
  _capitalize,
  _lowerFirst,
  _split,
  _upperFirst,
  removeWhitespace,
  resultToString,
} from './string.util'

test('capitalize', () => {
  expect(_capitalize('abc')).toBe('Abc')
  expect(_capitalize('aBc')).toBe('Abc')
})

test('upperFirst', () => {
  expect(_upperFirst('abc')).toBe('Abc')
  expect(_upperFirst('aBc')).toBe('ABc')
})

test('lowerFirst', () => {
  expect(_lowerFirst('Abc')).toBe('abc')
})

test('removeWhitespace', () => {
  expect(removeWhitespace(' 1 * A ')).toBe('1*A')
})

const anyItems = [
  undefined,
  null,
  '',
  'hello a',
  0,
  1,
  -5,
  () => 'smth',
  {},
  [],
  ['a'],
  { a: 'aa' },
  new Error('err msg'),
  new HttpError('http err msg', {
    httpStatusCode: 400,
  }),
]

test('resultToString', () => {
  expect(anyItems.map(resultToString)).toMatchSnapshot()
})

test('_split', () => {
  expect(_split('a b c', ' ', 1)).toEqual(['a b c'])
  expect(_split('a b c', ' ', 2)).toEqual(['a', 'b c'])
  expect(_split('a b c', ' ', 3)).toEqual(['a', 'b', 'c'])
})
