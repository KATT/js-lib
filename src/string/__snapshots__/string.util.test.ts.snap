// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resultToString 1`] = `
Array [
  "undefined",
  "null",
  "empty_string",
  "hello a",
  "0",
  "1",
  "-5",
  "() => 'smth'",
  "{}",
  "empty_string",
  "a",
  "{
  \\"a\\": \\"aa\\"
}",
  "err msg",
  "http err msg",
]
`;
